src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,43 +1,77 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [loader, setLoader] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(selectedTodo.userId).then(person => {
+      setUser(person);
+      setLoader(false);
+    });
+  });
```

Comment:
The useEffect hook should include 'selectedTodo.userId' as a dependency to avoid unnecessary re-renders and ensure the effect runs when 'selectedTodo.userId' changes.